### PR TITLE
Fix light saving

### PR DESCRIPTION
**Pull request by dog194**
### Description

Fix some bug with light.
now the light of RedstoneLamp, Furnace and LitPumpkin save properly.  

Fix some naming issue.

### Reason to modify

It should be

### Tests & Reviews

Tested

### CODE REVIEW

File: src/pocketmine/block/GlowingRedstoneOre.php [hunk 1]
```diff
@@ -24,12 +24,12 @@
...
 use pocketmine\level\Level;
 
-class GlowingRedstoneOre extends RedstoneOre{
+class GlowingRedstoneOre extends RedstoneOre implements SolidLight{
 
 	protected $id = self::GLOWING_REDSTONE_ORE;
 
 	public function getName() : string{
-		return "Glowing Redstone Ore";
+		return "Glowing RedStone Ore";
```
Comment (ishitatsuyuki) [inline src/pocketmine/block/GlowingRedstoneOre.php]: No, this is wrong.

File: src/pocketmine/block/Glowstone.php [hunk 2]
```diff
@@ -34,7 +34,7 @@ public function __construct(){
 	}
 
 	public function getName() : string{
-		return "Glowstone";
+		return "GlowStone";
```
Comment (ishitatsuyuki) [inline src/pocketmine/block/Glowstone.php]: This is also wrong.

File: src/pocketmine/block/RedstoneOre.php [hunk 3]
```diff
@@ -35,7 +35,7 @@ public function __construct(){
 	}
 
 	public function getName() : string{
-		return "Redstone Ore";
+		return "RedStone Ore";
```
Comment (ishitatsuyuki) [inline src/pocketmine/block/RedstoneOre.php]: Do not modify redundant locations.

File: src/pocketmine/block/Block.php [hunk 4]
```diff
@@ -321,14 +321,12 @@ public static function init(){
 
 					if($block->isSolid()){
 						if($block->isTransparent()){
-							if($block instanceof Lava){
-								self::$lightFilter[$id] = 15;
-							}elseif($block instanceof Liquid or $block instanceof Ice){
+							if($block instanceof Liquid or $block instanceof Ice){
```
Comment (ishitatsuyuki) [inline src/pocketmine/block/Block.php]: Lava will be likely to no longer emit light with this, please don't change.

**Comment by ishitatsuyuki** [comment]
This looks good, but there are some redundant changes. Squash your commits or tell me if you can't.

File: src/pocketmine/block/Block.php (hunk 4 above)
Comment (ishitatsuyuki) [inline src/pocketmine/block/Block.php]: Liquid is also not solid.

**Comment by dktapps** [comment]
Code Review -1. Reason: Furnace lighting does not update correctly when activated.
Also, there seems to be some lighting issues with lava in caves - see screenshot.
image

**Comment by dktapps** [comment]
This would also seem to be an issue.
image

**Comment by dog194** [comment]
@dktapps  now?

**Comment by dog194** [comment]
screenshot_2016-07-29-23-20-00
screenshot_2016-07-29-23-20-46
screenshot_2016-07-29-23-25-22
screenshot_2016-07-29-23-34-47
I cannot find a LAVA_LAKE, so i make one manually.
The light still have little different with single player.(Blame the computing way, i only fix the saving issue with RedstoneLamp, Furnace and LitPumpkin)
And the bug i fixing now blame ishitatsuyuki. He  Squash my commits and change my code in a wrong way. Also make me have more trouble to find the code I changed.

[Rejected with PullApprove](https://pullapprove.com/iTXTech/Genisys/pull-request/1555/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=iTXTech/Genisys)

**Comment by dktapps** [comment]
I'm also having difficulty finding a lava pool (the new caves generator is so inconsistent xD) but I can't find any other issues.

**Comment by PeratX** [comment]
Looks good! +1

**Pull request merged**